engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala

```diff
@@ -569,7 +569,9 @@ object ProgramExecutionSupport {
     */
   private def toMethodCall(value: ExpressionValue): Option[Api.MethodCall] =
     for {
-      call          <- Option(value.getCallInfo).orElse(Option(value.getCachedCallInfo))
+      call <-
+        if (Types.isPanic(value.getType)) Option(value.getCallInfo)
+        else Option(value.getCallInfo).orElse(Option(value.getCachedCallInfo))
```

Comment:
I know that this logic has been here already but it kind of looks weird that first we ask for call info and if missing we are asking for the cached value. I would expect the opposite to happen.

Reply:
This logic covers the case when the user comes back from the function (pops the execution frame). In this case, the expression does not have the call info because the result was cached and no function was called, but contains a cached call info.
I agree that it's weird but it always breaks some other cases when I try to come up with a workaround.